src/spv/bitcoin/BRPeer.cpp

```diff
@@ -1035,17 +934,10 @@ static void *_peerThreadRoutine(void *arg)
             len = 0;
 
             while (socket != INVALID_SOCKET && ! error && len < HEADER_LENGTH) {
-//                n = read(socket, &header[len], sizeof(header) - len);
                 n = recv(socket, (char*)&header[len], sizeof(header) - len, 0);
                 if (n > 0) len += n;
                 if (n == 0) error = ECONNRESET;
                 if (n < 0 && WSAGetLastError() != WSAEWOULDBLOCK) error = WSAGetLastError();
-                if (error == WSAENOTSOCK) {
```

Comment:
Is there a specific reason we're removing this? 

The situation I have in mind of occurring is if the OS is forced to closed a socket for whatever reason, and then this handle is reused for something else (in which case the right behaviour for us is indeed to ignore).

Reply:
Resolving, since worst case, it'll log error and as pointed out by Peter that this is closer to upstream.